src/productList/ProductList.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useEffect, useTransition } from 'react'
+import { Card, Input, Spin, Alert, AutoComplete } from 'antd'
+import { reqProductList, reqSearchProduct } from '../api'
+import './produc-list.less'
+const { Meta } = Card;
+const { Search } = Input;
+
+let cutArr = [] //用来存放切割后的数组
```

Comment:
You don't need to use this global variable. You can maintance both full product list and current product list in state and change current list based on full list.